### PR TITLE
add cmake package

### PR DESCRIPTION
**Pull request by dand-oss**
From your cmake project
-----------------------

find_package(verdigris CONFIG REQUIRED)

target_link_libraries(${libname} PUBLIC
   ...
   verdigris::verdigris
   )

Todo
----
- build and run tests and benchmarks

Build result
------------

[0/1] Install the project...
-- Install configuration: "Release"
-- Up-to-date: INSTALLDIR/share/verdigris/cmake/verdigrisTargets.cmake
-- Up-to-date: INSTALLDIR/share/verdigris/cmake/verdigrisConfig.cmake
-- Up-to-date: INSTALLDIR/share/verdigris/cmake/verdigrisConfigVersion.cmake
-- Installing: INSTALLDIR/include/verdigris/wobjectcpp.h
-- Installing: INSTALLDIR/include/verdigris/wobjectdefs.h
-- Installing: INSTALLDIR/include/verdigris/wobjectimpl.h

### CODE REVIEW

**Comment by ogoffart** [comment]
I think the problem with the CLA assistant is that the email used in the git commit doesn't match the email registered in the GitHub account.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+add_library(${libname}::${libname} ALIAS ${libname})
+
+include(GNUInstallDirs)
+
+target_include_directories(
+  ${libname}
+  INTERFACE $<BUILD_INTERFACE:${${libname}_SOURCE_DIR}/lib>
+            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+
+target_compile_features(${libname} INTERFACE cxx_std_11)
```
Comment (arBmind) [inline CMakeLists.txt:22]: I'd guess we need C++14 here.

**Review by arBmind — CHANGES_REQUESTED** [review]
@dand-oss Thank you for your PR.

Would it be possible/necessary to also express that Verdigris also requires a Qt5 or Qt6 to work as intended?

**Comment by arBmind** [comment]
Please note: #99 adds full CMake support to Verdigris. That contains install package building and running all test targets.

**Pull request closed without merging**